presto-benchmark-runner/src/main/java/com/facebook/presto/benchmark/framework/BenchmarkRunner.java

```diff
@@ -53,20 +57,30 @@ public void start()
         BenchmarkSuite benchmarkSuite = benchmarkSuiteSupplier.get();
         BenchmarkSuiteInfo benchmarkSuiteInfo = benchmarkSuite.getSuiteInfo();
         List<PhaseSpecification> phases = benchmarkSuiteInfo.getPhases();
+        int successfulPhases = 0;
 
         try {
             for (PhaseSpecification phase : phases) {
                 PhaseExecutor phaseExecutor = phaseExecutorFactory.get(phase, benchmarkSuite);
                 try {
-                    phaseExecutor.run();
+                    phaseExecutor.run(proceedWithFailures);
                     postEvent(BenchmarkPhaseEvent.succeeded(phase.getName()));
+                    successfulPhases++;
                 }
                 catch (PhaseExecutionException e) {
                     postEvent(BenchmarkPhaseEvent.failed(phase.getName(), e.toString()));
-                    throw new SuiteExecutionException();
+                    if (!proceedWithFailures) {
+                        throw new SuiteExecutionException();
+                    }
                 }
             }
-            postEvent(BenchmarkSuiteEvent.succeeded(benchmarkSuite.getName()));
+
+            if (successfulPhases < phases.size()) {
+                postEvent(BenchmarkSuiteEvent.completedWithFailures(benchmarkSuite.getName(), format("%s Phases out of %s succeeded", successfulPhases, phases.size())));
```

Comment:
Please make sure that you commits actually builds. I think `successfulPhases` has been removed from `BenchmarkSuiteEvent` so this should fail to build.

Reply:
I am not passing this `successfulPhases` to `BenchmarkSuiteEvent`. I am just using it to create an error message that is passed on create the `BenchmarkSuiteEvent` of type `COMPLETED_ON_FAILURES`